src/test/java/com/snowflake/kafka/connector/records/ConverterTest.java

```diff
@@ -258,6 +259,19 @@ public void testConnectJsonConverter_MapBigDecimalExceedsMaxPrecision() throws J
     //assert expected.toString().equals(result.toString());
   }
 
+  @Test
+  public void testConnectSimpleHeaderConverter_MapDateAndOtherTypes() throws JsonProcessingException {
```

Comment:
The merge gates are failing because the new code path is not executed.

Reply:
It's executed when I run locally, and when I revert the changes to `RecordService`, the test fails. There may be a difference between my local environment and the testing environment that's contributing to this.

Reply:
Is the testing environment using the newer version of the `connect-api` artifact?

Reply:
Yes, it is using newer version, I think issue is with timestamp conversion. Hence the expected result and actual results dont match. (6940800000 vs 6930000000L)

Should we put some date format?

Reply:
`Timestamp.fromLogical(schema, (java.util.Date) value));`
^ is the place where is it converting the value. May be in test we should explicitly provide Time Zone to avoid this issue or being dependent on the timeformat where the test runs.